src/Model/Client.php

```diff
@@ -119,11 +140,17 @@ public function setActive(bool $active): self
         return $this;
     }
 
+    /**
+     * @psalm-mutation-free
+     */
```

Comment:
How does psalm leverage this? I mean, what's the impact of using this annotation?

Reply:
this tells psalm that the method does result in side effects, allowing people using the bundle to use this method in a pure context.

e.g:
```
/**
 * @pure
 */
function foo(Client $client): string
{
  if ($client->isConfidential()) {
    return 'bar';
  }

  return 'baz';
}
```

If we remove the `@psalm-mutation-free` annotation, psalm will result in a error there as it is not sure if `isConfidentials` will result in side effects/mutation to `$client` or the global state of the application.

Reply:
also, in a mutation free method, you are not allowed to change property values, change global variables, or call any other mutating function/method.

https://psalm.dev/docs/annotating_code/supported_annotations/#psalm-mutation-free

Reply:
thank you!